adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/DatasourceManager.java

```diff
@@ -160,8 +162,10 @@ public void validate(Datasource datasource, DataImport dataImport) throws IOExce
     ValidationException e,
     RuntimeParameters runtimeParameters
   ) {
-    System.out.println("seting warning");
     dataImport.setHealth("WARNING");
+    Type listType = new TypeToken<List<String>>() {}.getType();
+    String errorMessagesAsString = new Gson().toJson(e.getAllMessages(), listType);
+    dataImport.setErrorMessages(errorMessagesAsString);
```

Comment:
Can we solve it somewhat more generic like in https://www.baeldung.com/java-hibernate-map-postgresql-array ?